ModelMetadataExtensions/ConventionalModelMetadataProvider.cs

```diff
@@ -38,16 +38,19 @@ public ConventionalModelMetadataProvider(bool requireConventionAttribute, Type d
 
             var defaultResourceType = DefaultResourceType;
             var conventionAttribute = conventionType.GetAttributeOnTypeOrAssembly<MetadataConventionsAttribute>();
-            if (conventionAttribute != null && conventionAttribute.ResourceType != null)
+            if (conventionAttribute != null)
             {
-                defaultResourceType = conventionAttribute.ResourceType;
+                if (conventionAttribute.ResourceType != null)
+                {
+                    defaultResourceType = conventionAttribute.ResourceType;
+                }
```

Comment:
This change is unnecessary. The C# `&&` operator is short-circuited  so if `conventionAttribute` is null, it won't evaluate the right hand side.